lib/streamlit/external/langchain/streamlit_callback_handler.py

```diff
@@ -199,6 +199,8 @@ def on_tool_start(
             label=self._labeler.get_tool_label(self._last_tool, is_complete=False),
             state="running",
         )
+        if len(input_str) > MAX_TOOL_INPUT_STR_LENGTH:
+            self._container.markdown(f"**Input:**\n\n{input_str}\n\n**Output:**")
```

Comment:
Is the `**Output:**` at the end of the string here with nothing after it because we expect the output itself to be printed in `on_tool_end`?

It may be preferable to move where "**Output:**  is printed to also be in `on_tool_end` as I think that'd be more clear.

Reply:
Oh, whoops, sorry it didn't quite register with me that that's what was originally done, but things were changed to only add the labels if the input exceeded a certain length.

In that case, I think it'd be sufficient to just leave a comment explaining that output is printed later and thus doesn't have to be included in the format string here